File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
     int logset(string s){
        unordered_set<char>charset;
        int right=0,left=0,max1=0;
        while(right<s.size()){
            if(charset.find(s[right])==charset.end()){
                charset.insert(s[right]);
                max1=max(max1,right-left+1);
                right++;
            }
            else{
                charset.erase(s[left]);
                left++;
            }
        }
        return max1;
     }
public:
    int lengthOfLongestSubstring(string s) {
        return logset(s);
    }
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        l=0
        seen=[]
        max_len=0
        for i in range(len(s)):
            while s[i] in seen:
                seen.remove(s[l])
                l+=1
            seen.append(s[i])
            max_len=max(max_len,i-l+1)
        return max_len
        